### PR TITLE
Speed up createNext test suite isolation

### PR DESCRIPTION
**Pull request by timneutkens**
## What?

Before: 25.71s

CleanShot 2024-04-23 at 12 19 42@2x

After: 11.05s (-57%)

CleanShot 2024-04-23 at 12 16 35@2x

## How?

Currently the system for isolation looks like this:

- Copy `packages` folder to an isolated directory
- Run `pnpm pack` for all folders in `packages`
- Collect the pack files, add them as `dependencies` in package.json of the isolated application
- Run `pnpm install`

Because the `next-swc` (Turbopack + SWC, yes we still need to rename the package) binary file is quite large in development (900MB+) it means we have to copy, then zip (pnpm pack), then unzip (pnpm install) that binary, which takes about 3+ seconds in each step.

The change in this PR is to skip the copy/zip/unzip completely by providing the folder path for the binary directly to Next.js, as it's a binary we don't need the special isolation for this, it's already standalone so running it directly allows us to skip 14,6 seconds of work that is required for each isolated test in development.

This will likely have little effect on CI times as we already do some tricks like only running the packing at the start of the CI process. Only thing that could be better and is probably worth doing is adopting this change for the time it saves for unzipping, that's almost 4 seconds per test still.

Closes NEXT-3200

### CODE REVIEW

**Comment by timneutkens** [comment]
* **#64909** <a href="https://app.graphite.dev/github/pr/vercel/next.js/64909?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `canary`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @timneutkens and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Comment by ijjk** [comment]
## Failing test suites 
Commit: 93a4d3851aa95cb25465b5fe7529aaf4e69d7b79

`pnpm test-start test/e2e/app-dir/app-compilation/index.test.ts`
- app dir > Loading > should render loading.js in initial html for slow page

<details>
<summary>Expand output</summary>

  ● app dir › Loading › should render loading.js in initial html for slow page

    next build failed with code/signal 1

      89 |           if (code || signal)
      90 |             reject(
    > 91 |               new Error(`next build failed with code/signal ${code || signal}`)
         |               ^
      92 |             )
      93 |           else resolve()
      94 |         })

      at ChildProcess.<anonymous> (lib/next-modes/next-start.ts:91:15)

</details>


Read more about building and testing Next.js in [contributing.md](https://github.com/vercel/next.js/blob/canary/contributing.md).

**Comment by ijjk** [comment]
## Failing test suites 
Commit: 93a4d3851aa95cb25465b5fe7529aaf4e69d7b79

`__NEXT_EXPERIMENTAL_PPR=true pnpm test-start test/e2e/app-dir/actions-allowed-origins/app-action-disallowed-origins.test.ts` (PPR)
- app-dir action disallowed origins > should error if x-forwarded-host does not match the origin

<details>
<summary>Expand output</summary>

  ● app-dir action disallowed origins › should error if x-forwarded-host does not match the origin

    next build failed with code/signal 1

      89 |           if (code || signal)
      90 |             reject(
    > 91 |               new Error(`next build failed with code/signal ${code || signal}`)
         |               ^
      92 |             )
      93 |           else resolve()
      94 |         })

      at ChildProcess.<anonymous> (lib/next-modes/next-start.ts:91:15)

</details>


Read more about building and testing Next.js in [contributing.md](https://github.com/vercel/next.js/blob/canary/contributing.md).

`__NEXT_EXPERIMENTAL_PPR=true pnpm test-start test/e2e/app-dir/app-basepath-custom-server/index.test.ts` (PPR)
- custom-app-server-action-redirect > redirects with basepath properly when server action handler uses `redirect`
- custom-app-server-action-redirect > redirects with proper cookies set from both redirect response and post respose

<details>
<summary>Expand output</summary>

  ● custom-app-server-action-redirect › redirects with basepath properly when server action handler uses `redirect`

    next build failed with code/signal 1

      89 |           if (code || signal)
      90 |             reject(
    > 91 |               new Error(`next build failed with code/signal ${code || signal}`)
         |               ^
      92 |             )
      93 |           else resolve()
      94 |         })

      at ChildProcess.<anonymous> (lib/next-modes/next-start.ts:91:15)

  ● custom-app-server-action-redirect › redirects with proper cookies set from both redirect response and post respose

    next build failed with code/signal 1

      89 |           if (code || signal)
      90 |             reject(
    > 91 |               new Error(`next build failed with code/signal ${code || signal}`)
         |               ^
      92 |             )
      93 |           else resolve()
      94 |         })

      at ChildProcess.<anonymous> (lib/next-modes/next-start.ts:91:15)

</details>


Read more about building and testing Next.js in [contributing.md](https://github.com/vercel/next.js/blob/canary/contributing.md).

File: packages/font/package.json [hunk 1]
```diff
@@ -25,6 +25,6 @@
     "fontkit": "2.0.2"
   },
   "peerDependencies": {
-    "next": "*"
+    "next": "14.3.0-canary.15"
```
Comment (timneutkens) [inline packages/font/package.json:28]: This was causing another version of Next.js to be installed, i.e. 14.2.1 and as such caused conflicts and sometimes resolved to the older version incorrectly.

**Review by huozhi — APPROVED** [review]
Nice PR!

**Review by ForsakenHarmony — APPROVED** [review]

File: .github/actions/next-stats-action/src/prepare/repo-setup.js [hunk 2]
```diff
@@ -79,162 +91,190 @@ module.exports = (actionInfo) => {
...
+              })
+              if (nextSwcVersion) {
+                Object.assign(packageJson.dependencies, {
+                  '@next/swc-linux-x64-gnu': nextSwcVersion,
+                })
+              } else {
+                if (nextSwcPkg) {
+                  // packageJson.dependencies['@next/swc'] =
+                  //   nextSwcPkg.packedPackageTarPath
+                }
```
Comment (ijjk) [inline .github/actions/next-stats-action/src/prepare/repo-setup.js:180]:
```suggestion
```
Looks like this could be deleted now

**Review by ijjk — APPROVED** [review]

**Pull request merged**